htlcswitch/switch.go

```diff
@@ -433,6 +454,11 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 			return err
 		}
 
+		// If this is failure than we need to obfuscate the error.
```

Comment:
It seems that a case above for the malformed onion error needs to be added. Otherwise, where is this handled currently in the PR? (still getting through it, so may be in the next few commits after this one).